Raise-FPE test failures on Windows and PPC (Trac #1755)

_Original ticket http://projects.scipy.org/numpy/ticket/1755 on 2011-03-04 by @rgommers, assigned to unknown._

np.power fails to raise an exception on Windows (tested on py-3.1 and py-2.6). Test failure:

```
======================================================================
FAIL: test_floating_exceptions (test_numeric.TestFloatExceptions)
Test basic arithmetic function errors
----------------------------------------------------------------------
Traceback (most recent call last):
  File "Z:\Users\rgommers\.wine\drive_c\Python31\lib\site-packages\numpy\core\tests\test_numeric.py", line 320, in test_floating_exceptions
    np.power, ftype(2), ftype(2**fi.nexp))
  File "Z:\Users\rgommers\.wine\drive_c\Python31\lib\site-packages\numpy\core\tests\test_numeric.py", line 261, in assert_raises_fpe
    "Type %s did not raise fpe error '%s'." % (ftype, fpeerr))
  File "Z:\Users\rgommers\.wine\drive_c\Python31\lib\site-packages\numpy\testing\utils.py", line 34, in assert_
    raise AssertionError(msg)
AssertionError: Type <class 'numpy.float64'> did not raise fpe error 'overflow'.
```

Result on Windows:

```
>>> 2.0 ** 2048
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: (34, 'Result too large')
>>> np.power(2.0, 2048)
inf
```

Result on OS X (as expected)

```
>>> 2.0**2048
------------------------------------------------------------
Traceback (most recent call last):
  File "<ipython console>", line 1, in <module>
OverflowError: (34, 'Result too large')

>>> np.power(2.0, 2048)
Warning: overflow encountered in power
inf
```

--- a/numpy/core/tests/test_numeric.py
+++ b/numpy/core/tests/test_numeric.py
@@ -552,7 +552,6 @@ class TestFloatExceptions(object):
         self.assert_raises_fpe(fpeerr, flop, sc1, sc2[()])
         self.assert_raises_fpe(fpeerr, flop, sc1[()], sc2[()])
 
-    @pytest.mark.xfail(reason="See ticket #2350")
     def test_floating_exceptions(self):
         # Test basic arithmetic function errors
         with np.errstate(all='raise'):

